src/system.h

```diff
@@ -41,6 +41,9 @@ struct Image {
     }
 
     wxBitmap &Display() {
+        // This might run in multiple threads in parallel
+        // so this function must not touch any global resources
+        // and callees must be thread-safe.
         if (!bm_display.IsOk()) {
             auto mapitem = imagetypes.find(image_type);
             if (mapitem == imagetypes.end()) return wxNullBitmap;

```

Comment:
below it accesses the global `sys`, though those functions do not appear to depend on `sys`. So maybe safer to make those function global so it is more obvious what they use.